Framework/DataHandling/src/LoadDetectorsGroupingFile.cpp

```diff
@@ -35,25 +35,41 @@ using namespace Mantid::API;
 namespace Mantid {
 namespace DataHandling {
 
+namespace {
+namespace PropertyNames {
+const std::string INPUT_FILE("InputFile");
+const std::string INPUT_WKSP("InputWorkspace");
+const std::string OUTPUT_WKSP("OutputWorkspace");
+} // namespace PropertyNames
+} // namespace
+
 DECLARE_ALGORITHM(LoadDetectorsGroupingFile)
 
 void LoadDetectorsGroupingFile::init() {
   /// Initialise the properties
 
   const std::vector<std::string> exts{".xml", ".map"};
-  declareProperty(Kernel::make_unique<FileProperty>("InputFile", "",
-                                                    FileProperty::Load, exts),
+  declareProperty(Kernel::make_unique<FileProperty>(
+                      PropertyNames::INPUT_FILE, "", FileProperty::Load, exts),
                   "The XML or Map file with full path.");
 
+  declareProperty(
```

Comment:
Would it be worth expanding on this a bit more in the documentation as to why you would want to use this `InputWorkspace` option? 

From our offline chat, the case I assume it fixes is where thegrouping file just has an instrument name so you'll always just get the most updated IDF for that instrument with the old version but with the `InputWorkspace` you can get the IDF for a time before the instrument changed in the workspace.